File: selenium/src/java/com/thoughtworks/selenium/internal/IndexFilterFunction.java
package com.thoughtworks.selenium.internal;

import java.util.Collections;
import java.util.List;

import com.thoughtworks.selenium.SeleniumException;

public class IndexFilterFunction implements FilterFunction {
	public List filterElements(List allElements, String filterValue) {
		try {
			int index = Integer.parseInt(filterValue);
			if (allElements.size() > index)
				return Collections.singletonList(allElements.get(index));
		} catch (NumberFormatException e) {
			// Handled below
		}
		
		throw new SeleniumException("Element with index " + filterValue + " not found");
	}
}

File: dotnet/src/Selenium.WebDriverBackedSelenium/Internal/SeleniumEmulation/FilterFunctions/IndexFilterFunction.cs
using System;
using System.Collections.Generic;
using System.Text;
using OpenQA.Selenium;

namespace Selenium.Internal.SeleniumEmulation.FilterFunctions
{
    /// <summary>
    /// Defines a function for filtering element lists by index.
    /// </summary>
    internal class IndexFilterFunction : IFilterFunction
    {
        #region IFilterFunction Members
        /// <summary>
        /// Filters a list of elements by the specified value.
        /// </summary>
        /// <param name="allElements">A list of <see cref="IWebElement"/> objects to be filtered.</param>
        /// <param name="filterValue">The value to filter on.</param>
        /// <returns>The filtered list of <see cref="IWebElement"/> objects.</returns>
        public IList<IWebElement> FilterElements(IList<IWebElement> allElements, string filterValue)
        {
            int index = -1;
            if (int.TryParse(filterValue, out index))
            {
                if (index < allElements.Count)
                {
                    return new List<IWebElement>() { allElements[index] };
                }
            }

            throw new SeleniumException("Element with index " + filterValue + " not found");
        }

        #endregion
    }
}
